salt/states/service.py

```diff
@@ -966,3 +966,40 @@ def mod_watch(name,
     ret['comment'] = 'Service {0}'.format(past_participle) if result else \
                      'Failed to {0} the service'.format(verb)
     return ret
+
+
+def systemctl_reloaded(name):
+    '''
+    Ensure that the systemctl daemon is reloaded.
+
+    .. note::
+        This state is only available on minions which use systemd.
+
+    name
+        Not used, the state_ID.
+
+    .. code-block:: yaml
+
+        systemd_daemon_is_reloaded:
+          service.systemctl_reloaded
+
+    '''
+    ret = {'name': name,
+           'changes': {},
+           'result': True,
+           'comment': 'Systemd is reloaded'}
+
+    if 'service.systemctl_reload' not in __salt__:
+        ret['result'] = False
+        ret['comment'] = "'systemctl daemon_reload' modules not available on this minion."
+    elif __opts__.get('test', False):
+        ret['result'] = None
+        ret['comment'] = 'Systemd would have been reloaded'
+    else:
```

Comment:
Reload is not a free operation for environments with many units. Can you detect if a reload is needed and only execute in that case?

Reply:
The user of this states will use the requisit onchanges. 

```yaml
service_file_is_managed:
  file.managed:
    - name: /etc/systemd/system/myservice.service
    - source: {{ 'salt://'~ slspath ~'/files/systemd/myservice.service.j2' }}
    - template: jinja
    - user: root
    - group: root
    - mode: 644

systemd_is_reloaded:
  leni_service.systemctl_reloaded:
    - onchanges:
      - file: service_file_is_managed
```

Is not simple to know if the systemd deamon have to be reloaded.